dodo/search.py

```diff
@@ -71,13 +73,17 @@ def refresh_thread(self, thread: QModelIndex|str):
         self.beginResetModel()
         self.d[row:row+1] = contents
         self.threads = {thread['thread']: i for i,thread in enumerate(self.d)}
+        self._count = len(self.d)
         self.endResetModel()
         logger.info("Model refreshed for '%s'", self.q)
 
-    def num_threads(self) -> int:
-        """The number of threads returned by the search"""
-
-        return len(self.d)
+    def num_threads(self, refresh_cache=False) -> int:
+        """The number of threads returned by the search."""
+        if refresh_cache:
+            r = subprocess.run(['notmuch', 'count', '--output=threads', self.q], capture_output=True)
+            if r.returncode == 0:
```

Comment:
Should something be logged at least if this fails? Seems like this could be very hard to debug if it breaks for some reason and just silently doesn't refresh the count anymore.

Reply:
Now logging stderr in warnings if something goes wrong.